esx_service/vmodl/VsphereContainerService.py

```diff
@@ -0,0 +1,400 @@
+
+   @JavaDocs(parent=_name, docs=
+   """
+   Add a datastore access privilege for this tenant.
+   @param privilege privilege to be assigned to this tenant.
+   @param switch_default_datastore If set to true, switch the default datastore
+          for this tenant. False by default.
+   """
+   )
+   @Method(parent=_name, wsdlName="AddPrivilege")
+   @Param(name="privilege", typ="vim.vcs.storage.DatastoreAccessPrivilege")
+   @Param(name="switch_default_datastore", typ="boolean", flags=F_OPTIONAL)
+   @Return(typ='void')
+   def AddPrivilege(self, privilege, switch_default_datastore=False):
+       pass
+
+   @JavaDocs(parent=_name, docs=
+   """
+   Modify datastore access privilege for this tenant. Datastore itself cannot be changed.
+   @param datastore Privilege settings on this datastore will be changed.
+   @param allow_create If set to true, this tenant is allowed to create and delete volumes on
+          the datastore. If false, the tenant is allowed to mount and unmount volumes only.
+   @param volume_max_size Maximum size (in MiB) of one single volume. Zero means unlimited size.
+   @param volume_total_size Total storage usage quota (in MiB) allowed on this datastore.
+          Zero means unlimited quota.
+   @throws vim.fault.NotFound If the given datastore does not exist for this tenant.
+   @throws vmodl.fault.InvalidArgument If the specified parameters are invalid,
+           e.g. volume total size exceeds the datastore capacity.
+   """
+   )
+   @Method(parent=_name, wsdlName="ModifyPrivilege",
+           faults=["vim.fault.NotFound", "vmodl.fault.InvalidArgument"])
+   @Param(name="datastore", typ="vim.Datastore")
+   @Param(name="allow_create", typ="boolean", flags=F_OPTIONAL)
+   @Param(name="volume_max_size", typ="int", flags=F_OPTIONAL)
+   @Param(name="volume_total_size", typ="int", flags=F_OPTIONAL)
+   @Return(typ='void')
+   def ModifyPrivilege(self, datastore, allow_create, volume_max_size, volume_total_size):
+       pass
+
+   @JavaDocs(parent=_name, docs=
+   """
+   Remove access privilege on the given datastore from this tenant.
+   @param datastore Privilege on this datastore will be removed from this tenant.
+   """
+   )
+   @Method(parent=_name, wsdlName="RemovePrivilege")
+   @Param(name="datastore", typ="vim.Datastore")
+   @Return(typ='void')
+   def RemovePrivilege(self, datastore):
+       pass
+
+   @JavaDocs(parent=_name, docs=
+   """
+   Get privilege of this tenant on the given datastore.
+   @datastore If set, return privilege for this datastore only; otherwise return all privileges.
+   @return Privilege of this tenant on the given datastore.
+   @throw NotFound If this tenant has no privilege on the given datastore.
+   """
+   )
+   @Method(parent=_name, wsdlName="GetPrivileges",
+        faults=["vim.fault.NotFound"])
+   @Param(name="datastore", typ="vim.Datastore", flags=F_OPTIONAL)
+   @Return(typ='vim.vcs.storage.DatastoreAccessPrivilege[]')
+   def GetPrivileges(self, datastore):
```

Comment:
I'm actually following existing VSAN VMODL style. Basically, if the return type is an array, the name will be plural, for example: GetTenants, GetVMs, GetPrivileges. Or if the input parameter is an array, the name is also plural: AddVMs, RemoveVMs, ReplaceVMs.

Please let me know if you still have concern about this. I can change it in the next PR if needed.